model/src/main/java/org/openremote/model/asset/AssetResource.java

```diff
@@ -277,4 +279,21 @@ void deleteAllUserAssetLinks(@BeanParam RequestParams requestParams,
     @Consumes(APPLICATION_JSON)
     @Produces(APPLICATION_JSON)
     Asset<?>[] queryAssets(@BeanParam RequestParams requestParams, AssetQuery query);
+
+    /**
+     * Change parent for a set of asset
+     */
+    @PUT
+    @Path("{parentAssetId}/child")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    void updateParent(@BeanParam RequestParams requestParams, @PathParam("parentAssetId") @NotNull(message = "Parent reference required") String parentId, @QueryParam("assetIds") @Size(min = 1, message = "At least one child to update parent reference") List<String> assetIds);
+
+    /**
+     * Remove parent reference from each asset referenced in the query parameter assetIds
+     */
+    @DELETE
+    @Path("/parent")
+    @Produces(APPLICATION_JSON)
+    void updateNoneParent(@BeanParam RequestParams requestParams, @QueryParam("assetIds") @Size(min = 1, message = "At least one child to update parent reference") List<String> assetIds);
 }
```

Comment:
Is there a reason for 2 methods rather than one?

Reply:
first is to attach to one parent new children.
second is the case where asset are put on root level of the asset tree (and so no parent and so no parent asset ID provided). I tried to avoid to put "undefined" or any strange value on the first PUT.